packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -4204,4 +4250,14 @@ export class SnapController extends BaseController<
       runtime.state = undefined;
     }
   }
+
+  /**
+   * Get Snap metadata from the current state without refreshing the registry database.
+   *
+   * @param snapId - A Snap id.
+   * @returns The available metadata for the Snap, if any.
+   */
+  #getSnapMetadata(snapId: SnapId) {
+    return this.#verifiedSnaps?.[snapId]?.metadata;
```

Comment:
This data is available in the registry, we don't need to duplicate it to `SnapController`